### PR TITLE
ARROW-4645: [C++/Packaging] Ship Gandiva with OSX and Windows wheels

### PR DESCRIPTION
**Pull request by kszucs**
crossbow builds:
- [kszucs/crossbow/build-448](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-448)

### CODE REVIEW

**Comment by wesm** [comment]
Code looks good, but the crossbow tasks fail

**Comment by wesm** [comment]
Maybe so. How are the Windows and macOS wheels for numba built?

**Comment by xhochy** [comment]
There are special conda recipes in `llvmlite` that don't use the defaults/conda-forger compiler toolchains: https://github.com/numba/llvmlite/tree/master/conda-recipes

**Comment by kszucs** [comment]
Can We depend on the llvmlite wheel as well?

**Comment by wesm** [comment]
llvmlite doesn't ship the "raw" LLVM libraries so I don't think that helps

**Comment by kszucs** [comment]
OSX wheels working now, I'm trying llvmdev for windows.

**Comment by kszucs** [comment]
Crossbow builds: [kszucs/crossbow/build-488](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-488)

**Comment by kszucs** [comment]
[kszucs/crossbow/build-493](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-493)

File: dev/tasks/python-wheels/travis.osx.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ before_install:
   - brew uninstall boost cgal postgis sfcgal
   - brew update
   - brew upgrade cmake
-  - brew install bison flex
+  - brew install bison flex llvm@7
```
Comment (xhochy) [inline dev/tasks/python-wheels/travis.osx.yml]: Does this link to the static libs then? We should not depend on `brew` packages being installed.

Comment (kszucs) [reply dev/tasks/python-wheels/travis.osx.yml]: AFAIK We link llvm statically in all cases. I'll remove llvm in the test section to check this.

File: dev/tasks/tasks.yml [hunk 2]
```diff
@@ -193,6 +196,7 @@ tasks:
       unicode_width: 16
       test_docker_images:
         - python:3.7-slim
+        - centos/python-37-centos7
```
Comment (kszucs) [inline dev/tasks/tasks.yml]: This image doesn't exists, that's why the corresponding crossbow task fails, removing it.

**Comment by kszucs** [comment]
[kszucs/crossbow/build-496](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-496)

**Comment by kszucs** [comment]
[kszucs/crossbow/build-502](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-502)

**Comment by kszucs** [comment]
@kou the build should pass now [kszucs/crossbow/build-523](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-523)

**Comment by kszucs** [comment]
[kszucs/crossbow/build-530](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-530)

**Comment by kszucs** [comment]
[kszucs/crossbow/build-536](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-536)

**Review by wesm — APPROVED** [review]
+1, thank you @kszucs and @xhochy, this looks like it was very painful, but hopefully things will be smoother from here on out =)

**Comment by kou** [comment]
wheel-win-cp35m and wheel-win-cp36m on AppVeyor are failed:
* https://ci.appveyor.com/project/kou/crossbow/builds/23383931
* https://ci.appveyor.com/project/kou/crossbow/builds/23383928

I don't look into them yet but they may be related to this.

**Comment by kou** [comment]
https://issues.apache.org/jira/browse/ARROW-5025

**Pull request closed without merging**